Show indication of changed lines in code overview

bobsilverberg opened the pull request:
Fixes #920 

This adds an indication for inserted lines into the `CodeOverview` component.

bobsilverberg commented:
@willdurand Here's a patch for our discussion later this morning.

bobsilverberg commented:
@willdurand This is ready for an actual review now.

willdurand commented:
>  The diff coverage is `97.22%`.

😮

bobsilverberg commented:
> > The diff coverage is `97.22%`.
> 
> 😮

Thanks for pointing that out. I'll take a look at the report to see if I can find the missing coverage.

bobsilverberg commented:
The coverage report looks like it was for the previous iteration, as I can see the code still has `console.log` calls in it. The missing coverage it calls out in `CodeOverview` may still be valid, so I'll check that. Doesn't it usually generate a new coverage report when commits are changed?

willdurand commented:
> Doesn't it usually generate a new coverage report when commits are changed?

it should

bobsilverberg commented on src/components/CodeOverview/index.tsx:
```diff
@@ -222,23 +236,29 @@ export class CodeOverviewBase extends React.Component<Props, State> {
-        )[0];
-        if (firstMsg) {
-          linkableLine = firstMsg.line;
+        const firstChange = shapes.filter((s) => insertedLines[s.line])[0];
```
It is possible, but when you reference an index that doesn't exist on an array you get `undefined`, and that's dealt with on the next line that checks `if (firstChange)`.

bobsilverberg commented:
@willdurand The missing code coverage actually uncovered a bug! I hadn't written a test for one particular case, and it turns out that case didn't actually work. I've added the coverage and fixed the bug now.
